testing/test-nx-utils/src/lib/utils/tree.ts

```diff
@@ -0,0 +1,30 @@
+import { Tree } from '@nx/devkit';
+import { mkdir, writeFile } from 'node:fs/promises';
+import { dirname, join } from 'node:path';
+
+async function ensureDirectoryExists(baseDir: string) {
+  try {
+    await mkdir(baseDir, { recursive: true });
+    return;
+  } catch (error) {
+    console.error((error as { code: string; message: string }).message);
+    if ((error as { code: string }).code !== 'EEXIST') {
+      throw error;
+    }
+  }
+}
+
+export async function materializeTree(tree: Tree, targetFolder: string) {
```

Comment:
This is a first a new helper to unlock a new way of generating a e2e setup. See it un use here: